compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -1318,13 +1319,36 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
                         && let hir::IsAsync::NotAsync = self.tcx.asyncness(lifetime_ref.hir_id.owner.def_id)
                         && !self.tcx.features().anonymous_lifetime_in_impl_trait
                     {
-                        rustc_session::parse::feature_err(
-                            &self.tcx.sess.parse_sess,
-                            sym::anonymous_lifetime_in_impl_trait,
-                            lifetime_ref.span,
-                            "anonymous lifetimes in `impl Trait` are unstable",
-                        ).emit();
-                        return;
+
+                        match self.tcx.hir().get_generics(lifetime_ref.hir_id.owner.to_def_id().as_local().unwrap()) {
+                            Some(generics) => {
```

Comment:
Now you have `generics` you want 2 things to know where to suggest the lifetime parameter `param_span`, and in what form `param_sugg`.
Once you have that, you call
```
diag.multipart_suggestion(
  "consider introducing a named lifetime parameter",
  vec![
    (lifetime_ref.span, "'a".to_owned()), // this is at the use site
    (param_span, param_sugg), // this is in generics
  ],
)
```

Reply:
it seems if I put "'a" for the message, it replaces the ampersand symbol from the original posters input. Should I keep this?

Reply:
You can use `lifetime_ref.span.shrink_to_hi()` instead then.